src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -283,7 +283,7 @@ export function uiRenderMixin(kbnServer, server, config) {
       },
     });
 
-    const csp = createCSPRuleString(config.get('csp.rules'));
+    const csp = createCspDirectives(config.get('csp.rules'));
```

Comment:
How will this be accessible in NP? Should we expose it from the http service setup contract?

Reply:
I don't think that is clear yet, and I think it is going to be solved at least in part via #50644.

Reply:
I think the ideal end-state here is that plugins never need to consume or care CSP headers. The HTML rendering service (https://github.com/elastic/kibana/issues/41964) should add this header under the hood in the future.